lib/constants.js

```diff
@@ -908,18 +909,11 @@ const unfold = (f, seed) => {
 }`
 
 if (typeof window !== 'undefined' && typeof window.navigator !== 'undefined') {
-  const alreadyLoaded = new Set()
+  require(`codemirror/mode/javascript/javascript`)
 
-  LANGUAGES.filter(language => language.mode !== 'auto' && language.mode !== 'text').forEach(
-    language => {
-      if (language.mode && !alreadyLoaded.has(language.mode)) {
-        alreadyLoaded.add(language.mode)
-        language.custom
-          ? require(`./custom/modes/${language.mode}`)
-          : require(`codemirror/mode/${language.mode}/${language.mode}`)
-      }
-    }
-  )
+  const alreadyLoaded = new Set(['javascript'])
+
+  dynamic(() => import('./modes')(alreadyLoaded))
```

Comment:
I don't think you need to pass in `alreadyLoaded` you can just do:
```js
import('./modes').then(modes => {
  modes.forEach(l => alreadyLoaded.add(l.mode))
})
```
or something similar

Reply:
or rather wrap the whole import call in a check to see it it had been imported before.